Shared/mods/deathmatch/logic/luadefs/CLuaCryptDefs.cpp

```diff
@@ -77,100 +78,75 @@ std::string CLuaCryptDefs::Base64decode(std::string str)
-                            [password, salt = options["salt"], cost] {
-                                // Execute time-consuming task
-                                return SharedUtil::BcryptHash(password, salt, cost);
-                            },
-                            [luaFunctionRef](const SString& hash) {
-                                CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaFunctionRef.GetLuaVM());
-                                if (pLuaMain)
-                                {
-                                    CLuaArguments arguments;
-
-                                    if (hash.empty())
-                                    {
-                                        m_pScriptDebugging->LogCustom(pLuaMain->GetVM(), "Invalid value for field 'salt'");
-                                        arguments.PushBoolean(false);
-                                    }
-                                    else
-                                        arguments.PushString(hash);
+                    CLuaShared::GetAsyncTaskScheduler()->PushTask<SString>(
+                        [password, salt = options["salt"], cost] {
+                            // Execute time-consuming task
+                            return SharedUtil::BcryptHash(password, salt, cost);
+                        },
+                        [luaFunctionRef = callback.value()](const SString& hash) {
+                            CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine(luaFunctionRef.GetLuaVM());
+                            if (pLuaMain)
+                            {
+                                CLuaArguments arguments;
 
-                                    arguments.Call(pLuaMain, luaFunctionRef);
+                                if (hash.empty())
+                                {
+                                    m_pScriptDebugging->LogCustom(pLuaMain->GetVM(), "Invalid value for field 'salt'");
+                                    arguments.PushBoolean(false);
                                 }
-                            });
+                                else
+                                    arguments.PushString(hash);
 
-                        lua_pushboolean(luaVM, true);
-                        return 1;
-                    }
+                                arguments.Call(pLuaMain, luaFunctionRef);
+                            }
+                        });
                 }
             }
             else
-                m_pScriptDebugging->LogWarning(luaVM, "Invalid value for field 'cost'");
+            {
+                // Sync
+                SString hash = SharedUtil::BcryptHash(password, options["salt"], cost);
+                if (!hash.empty())
+                {
+                    return hash;
+                }
+                else
+                    throw std::invalid_argument("Invalid value for field 'salt'");
```

Comment:
It might also return "", if cost is too big (22 seems like the limit).
Ideally we would throw in `BcryptHash` based on `errno` `brcypt` set, but lets leave that for a future PR.